config/initializers/doorkeeper.rb

```diff
@@ -24,8 +24,9 @@ def self.optional
   end
 
   resource_owner_from_credentials do
-    if params[:display_name] && u = User.find_for_database_authentication(display_name: params[:display_name])
-      valid_non_disabled_user = u.valid_password?(params[:password]) && !u.disabled?
+    if params[:display_name] && u = ZooniverseUser.authenticate(params[:display_name],
```

Comment:
No it would be testing against the User database, which may not be in sync with the ZooHome database (user just reset their password on ZooHome for example).